### PR TITLE
Specify address validating and binding

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: specification/identity_service_api.rst [hunk 1]
```diff
@@ -64,6 +65,108 @@ Association Lookup
...
+-------------------------
+
+The flow for creating an association is session-based.
+
+Within a session, one may prove that one has ownership of a 3pid.
+Once this has been established, the user can form an association between that
+3pid and a Matrix user ID. Note that this association is only proved one way;
+a user can associate *any* Matrix user ID with a validated 3pid,
+i.e. I can claim that any email address I own is associated with
+billg@microsoft.com.
```
Comment (dbkr) [inline specification/identity_service_api.rst]: presumably this should be @billg:microsoft.com

Comment (illicitonion) [inline specification/identity_service_api.rst]: Done

File: specification/identity_service_api.rst [hunk 2]
```diff
@@ -64,6 +65,108 @@ Association Lookup
...
+Creating a session
+~~~~~~~~~~~~~~~~~~
+
+A client makes a call to::
+
+ POST https://my.id.server:8090/_matrix/identity/v1/api/validate/email/requestToken
+
+ client_secret=monkeys_are_GREAT&
+ email=foo@bar.com&
+ send_attempt=1
```
Comment (dbkr) [inline specification/identity_service_api.rst]: Might be worth doccing send_attempt (increment it to actually send another email vs retry the http hit)

File: specification/identity_service_api.rst [hunk 3]
```diff
@@ -64,6 +65,108 @@ Association Lookup
...
+able to read the email for that email address, and so we validate ownership of
+the email address.
+
+We return the ``sid`` generated for this session to the caller, in a JSON object
+containing the ``sid`` key.
+
+Validating ownership of an email
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A user make make either a ``GET`` or a ``POST`` request to
```
Comment (dbkr) [inline specification/identity_service_api.rst]: may make?

File: specification/identity_service_api.rst [hunk 4]
```diff
@@ -64,6 +65,108 @@ Association Lookup
...
+ {"errcode": "M_SESSION_NOT_VALIDATED", "error": "This validation session has not yet been completed"}
+
+If the ``sid`` and ``client_secret`` were not recognised, or were not correct,
+an error will be returned.
+
+Publishing a validated association
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An association between a session and a Matrix user ID can be published by making
+an URL-encoded HTTP POST request to ``/_matrix/identity/v1/api/3pid/bind`` with
```
Comment (dbkr) [inline specification/identity_service_api.rst]: 'a' URL...?

**Comment by dbkr** [comment]
otherwise lgtm

File: specification/identity_service_api.rst (hunk 2 above)
Comment (illicitonion) [inline specification/identity_service_api.rst]: Done

File: specification/identity_service_api.rst (hunk 3 above)
Comment (illicitonion) [inline specification/identity_service_api.rst]: Done

File: specification/identity_service_api.rst (hunk 4 above)
Comment (illicitonion) [inline specification/identity_service_api.rst]: Done

**Pull request merged**